### PR TITLE
add xzip rule

### PR DESCRIPTION
**Pull request by mr-tz**

### CODE REVIEW

File: nursery/linked-against-xzip.yml [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+      - string: "ct_init: length != 256"
+      - string: "ct_init: dist != 256"
+      - string: "ct_init: 256+dist != 512"
+      - string: "bit length overflow"
+      - string: "code %d bits %d->%d"
+      - string: "inconsistent bit counts"
+      - string: "gen_codes: max_code %d "
+      - string: "dyn trees: dyn %ld, stat %ld"
+      - string: "bad pack level"
+      - string: "Code too clever"
```
Comment (mr-tz) [inline nursery/linked-against-xzip.yml:22]: my favorite string

File: nursery/linked-against-xzip.yml [hunk 2]
```diff
@@ -0,0 +1,30 @@
+rule:
+  meta:
+    name: linked against XZip
+    namespace: linking/static/xzib
```
Comment (williballenthin) [inline nursery/linked-against-xzip.yml:4]: typo?
```suggestion
    namespace: linking/static/xzip
```

File: nursery/linked-against-xzip.yml [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+      - string: "ct_init: 256+dist != 512"
+      - string: "bit length overflow"
+      - string: "code %d bits %d->%d"
+      - string: "inconsistent bit counts"
+      - string: "gen_codes: max_code %d "
+      - string: "dyn trees: dyn %ld, stat %ld"
+      - string: "bad pack level"
+      - string: "Code too clever"
+      - string: "unknown zip result code"
+      - string: "Culdn't duplicate handle"  # typo in library code
```
Comment (williballenthin) [inline nursery/linked-against-xzip.yml:24]: thanks for confirming the typo

File: nursery/linked-against-xzip.yml [hunk 4]
```diff
@@ -0,0 +1,30 @@
...
+      - string: "File not found in the zipfile"
+      - string: "Still more data to unzip"
+      - string: "Caller: the file had already been partially unzipped"
+      - string: "Caller: can only get memory of a memory zipfile"
+      - string: "Zip-bug: internal initialisation not completed"
+      - string: "Zip-bug: an internal error during flation"
```
Comment (williballenthin) [inline nursery/linked-against-xzip.yml:30]: i like this verb

**Review by williballenthin — APPROVED** [review]
one typo, otherwise good to go

**Comment by mr-tz** [comment]
good catch, thanks!

**Pull request merged**